test/powershell/Language/Scripting/DeserializedMethods.Tests.ps1

```diff
@@ -0,0 +1,27 @@
+﻿Describe "DeserializedMethods" -Tags "CI" {    
+    It "Deserialized objects shouldn't ever have any methods (unless they are primitive known types)" {
+        $a = new-object collections.arraylist
+        $null = $a.Add(1)
+        $null = $a.Add(2)
+        $null = $a.Add(3)
+        
+	    # using linkedlist that implements IEnumerable,
+	    # but doesn't implement IList or IList<T>
+	    $x = new-object collections.generic.linkedlist[int]
+	    $null = $x.Add(123)
+	    $null = $x.Add(456)
+
+        $s = [System.Management.Automation.PSSerializer]::Serialize($x)
+        $d = [System.Management.Automation.PSSerializer]::Deserialize($s)	    
+	    
+	    try
+	    {
+		    $null = $d.Add(789)
```

Comment:
We could test this without exceptions by using `Get-Member -MemberType *Method* Add | Should Be $null` or something like that.